### ISSUE TITLE
NullArgument false-positive on ArrayCollection

### ISSUE THREAD

**Issue by fluffycondor** [weirdan/doctrine-psalm-plugin#66]
Psalm 3.14.2@3538fe1955d47f6ee926c0769d71af6db08aa488 + weirdan/doctrine-psalm-plugin

> ERROR: NullArgument - src/Entity/Organization.php:155:13 - Argument 1 of Doctrine\Common\Collections\ArrayCollection::offsetSet cannot be null, null value provided to parameter with type int|string (see https://psalm.dev/057)
            $this->staff[] = $staff;

But offsetSet can be null:
```php
    /**
     * Required by interface ArrayAccess.
     *
     * {@inheritDoc}
     */
    public function offsetSet($offset, $value)
    {
        if (! isset($offset)) {
            $this->add($value);

            return;
        }

        $this->set($offset, $value);
    }
```

If I turn off the plugin, the errors are disappeared.
